tableauserverclient/server/endpoint/jobs_endpoint.py

```diff
@@ -48,3 +54,35 @@ def get_by_id(self, job_id):
         server_response = self.get_request(url)
         new_job = JobItem.from_response(server_response.content, self.parent_srv.namespace)[0]
         return new_job
+
+    def wait_for_job(self, job_id, *, timeout=None):
+        id_ = getattr(job_id, "id", job_id)
```

Comment:
If the caller is always passing in the string job_id, this check will always fail and return the job_id.

If you want to allow callers to pass in a JobItem OR a job_id, this is a clever way to do it but it's a little non-obvious at first read. You could call the parameter just `job` or `job_or_job_id` to help.

Reply:
I simply copied this trick from the `jobs.cancel` endpoint and saw it in many other endpoints in TSC.
I thought, it would be considered best practice...

Given this is consistent with the remaining code of TSC, do think I should change it here?

Reply:
It looks like all instances of that trick are related to jobs/extract refresh. I don't think I wrote any of those endpoints, which might be why I'm unfamiliar with the pattern. I'm not a huge fan but I guess it's already in there.

Reply:
@t8y8 I wrote that getattr trick on the refresh endpoints. The reason for it was because the datasource endpoint was expecint a `DatasourceItem` whereas the workbook endpoint was expecting a string of `WorkbookItem.id`. That trick allowed for a consistent, but backwards compatible endpoint API, where they both accepted both. (One of the reasons I ultimately started #804 )

Reply:
So what is our strategy going forward? Do we only want to accept strings? Or only `JobItem`? Or both?
If both: Do we want to use `getattr` or do we want to use a more specific typing, e.g.

```
if isinstance(job_id, JobItem):
    job_id = job_id.id
assert isinstance(job_id, str)
```

Reply:
If both, explicit `isinstance` checking would be better as it will play nicer with mypy if and when type hinting is implemented.

Reply:
ok, so it's `isinstance` now